tools/convertUsageStatsLogFile.php

```diff
@@ -0,0 +1,570 @@
+        }
+
+        if (empty($contextPaths) || !$page || !$operation) {
+            echo "Either context paths, page or operation could not be parsed from the URL correctly.\n";
+            return $noMatchesReturner;
+        }
+
+        $pageAndOperation = $page . '/' . $operation;
+
+        $pageAndOpMatch = false;
+        foreach ($expectedPageAndOp as $workingAssocType => $workingPageAndOps) {
+            foreach ($workingPageAndOps as $workingPageAndOp) {
+                if ($pageAndOperation == $workingPageAndOp) {
+                    // Expected url, don't look any futher.
+                    $pageAndOpMatch = true;
+                    break 2;
+                }
+            }
+        }
+        if ($pageAndOpMatch) {
+            return [$workingAssocType, $contextPaths, $page, $operation, $args];
+        } else {
+            echo "No matching page and operation found.\n";
+            return $noMatchesReturner;
+        }
+    }
+
+    /**
+    * Get the expected page and operation.
+    * They are grouped by the object type constant that
+    * they give access to.
+    */
+    protected function getExpectedPageAndOp(): array
+    {
+        $pageAndOp = [
+            Application::getContextAssocType() => [
+                'index/index'
+            ]
+        ];
+        $application = Application::get();
+        $applicationName = $application->getName();
+        switch ($applicationName) {
+            case 'ojs2':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'article/download'],
+                    Application::ASSOC_TYPE_SUBMISSION => [
+                        'article/view'],
+                    Application::ASSOC_TYPE_ISSUE => [
+                        'issue/view'],
+                    Application::ASSOC_TYPE_ISSUE_GALLEY => [
+                        'issue/download']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'index';
+                break;
+            case 'omp':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'catalog/download'],
+                    Application::ASSOC_TYPE_MONOGRAPH => [
+                        'catalog/book'],
+                    Application::ASSOC_TYPE_SERIES => [
+                        'catalog/series']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'catalog/index';
+                break;
+            case 'ops':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'preprint/download'],
+                    Application::ASSOC_TYPE_SUBMISSION => [
+                        'preprint/view']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'index';
+                break;
+        }
```

Comment:
Please add:
```php
`default: throw new Exception('Unrecognized application name.');
```
This will help us find affected areas if we ever get around to https://github.com/pkp/pkp-lib/issues/3030.